common/config/azure-pipelines/templates/run-eslint-deprecation.yaml

```diff
@@ -0,0 +1,64 @@
+      echo "##vso[task.setvariable variable=version;isOutput=true]$newPackageVersion"
+    displayName: Change addVersion in ESLint config to current version
+    name: getNewVersion
+
+  - bash: node common/scripts/install-run-rush.js lint-deprecation
+    displayName: rush lint-deprecation
+
+  - bash: git add .
+    displayName: git add .
+
+  - bash: |
+      git commit -m "Apply deprecation date rule for v$(getNewVersion.version)"
+    displayName: Commit deprecation comment changes to release branch
+
+  - bash: |
+      deprecationDateCommitHash=$(git log -1 --format=%H)
+      echo "##vso[task.setvariable variable=hash;isOutput=true]$deprecationDateCommitHash"
+    displayName: Save SHA-1 hash of the commit
+    name: deprecationCommit
+
+  - bash: |
+      git fetch https://$(GITHUBTOKEN)@github.com/iTwin/itwinjs-core master:master
+      git checkout master
+    displayName: Checkout to master
+
+  - bash: |
+      commitHash=$(deprecationCommit.hash)
+      echo "Attempting to cherry-pick commit: $commitHash"
+      if git cherry-pick $commitHash; then
+        git push https://$(GITHUBTOKEN)@github.com/iTwin/itwinjs-core HEAD:master
+        echo "Successfully applied deprecation comment updates to master"
+      else
+        # Merge conflicts: Create branch, set up PR for manual resolution.
+        echo "Cherry-pick operation on master failed (most likely due to merge conflicts). Manual intervention required"
+        git cherry-pick --abort
+
+        # Create and checkout to conflict handling branch
+        conflictBranchName="cherry-pick-conflict-$(date +%y%m%d%H%m%s)"
+        echo "Creating new branch '$conflictBranchName' for manual resolution"
+        git checkout -b $conflictBranchName
+
+        # Cherry-pick changes onto conflict handling branch
+        git cherry-pick $commitHash
+
+        # Commit and push all changes
+        git add .
+        git commit -m "Add deprecation rule changes with all merge conflicts"
+        git push --set-upstream https://$(GITHUBTOKEN)@github.com/iTwin/itwinjs-core $conflictBranchName
+        echo "Now manually fix conflicts on branch '$conflictBranchName' and merge changes into master"
+
+        # Create a PR with new conflict branch
+        gh pr create --title "Merge deprecation comment changes into master" --body "This PR contains all changes after running lint-deprecation ESLint rule and merging with master. Manual intervention required: solve conflicts in this branch and merge PR into master. This happened because deprecation comments in master and in release branch could not be merged automatically." --draft --head $conflictBranchName
+      fi
+    displayName: Perform cherry-pick operation for deprecation changes
+    continueOnError: true
+    env:
+      GH_TOKEN: $(GITHUBTOKEN)
+
+  - bash: git checkout $(deprecationCommit.hash)
+    displayName: Checkout back to branch that ran the pipeline
```

Comment:
Ignore the fact this is on line 64, I just needed to create a thread so I picked a random line.

Anyway... Looks like finalize release ran on the wrong branch here?
![image](https://github.com/user-attachments/assets/9b0fd361-2a67-46cf-8757-6a10ba0618f3)

Reply:
not sure how this fork is set up. But the finalize release workflow is triggered by the release branch getting bumped. And then it pushes those changes to the next major release, or master branch. Its made to cascade up all active release branches until it hits master. Ex
new patch on 4.5.5 creates change logs
Finalize release triggered on release/4.5.x
push change logs to release/5.0.x
Finalize release is triggered on release/5.0.x
push change logs to master